### PR TITLE
LPS-196630 [Marketplace] Implementation of license selection for DXP apps free

### PR DESCRIPTION
**Pull request by juliocezar96**
[LPS-196630](https://liferay.atlassian.net/browse/LPS-196630)

### CODE REVIEW

**Comment by juliocezar96** [comment]
## :fire: GitRay Report 
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
 ## Jira Tasks 

 <html><details><summary>More details about: <b>LPS-196630 [FE] Implementation of license selection for DXP apps free</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-196630">[FE] Implementation of license selection for DXP apps free</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Low</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Julio Cezar</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Julio Cezar</td>
        </tr>
                </table>
          </details><details><summary>More details about: <b>LPS-195705 [FE] Create License Selection home</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-195705">[FE] Create License Selection home</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Low</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Julio Cesar de Oliveira Guimaraes</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Jair Medeiros Ferreira Filho</td>
        </tr>
                </table>
          </details><details><summary>More details about: <b>LPS-195699 [FE] Create page-definition and change layout structure to a page</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-195699">[FE] Create page-definition and change layout structure to a page</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Low</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Peer Review</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Bruno Martins Gomes de Santana</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Jair Medeiros Ferreira Filho</td>
        </tr>
                </table>
          </details><details><summary>More details about: <b>LPS-196079 [FE] Create Radio Card Selector component</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-196079">[FE] Create Radio Card Selector component</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Low</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Peer Review</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Joao Marcelo Alves</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Bruno Martins Gomes de Santana</td>
        </tr>
                </table>
          </details></html> 

---

Hey, don't forget to update all related tickets with correct status, such as **In Review** :wink: 
### Git Stats 

824 files changed, 5156 insertions(+), 18768 deletions(-)

This PR is too huge for one to review :broken_heart: 

Batman

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const getSiteURL = () => {
+	const layoutRelativeURL = Liferay.ThemeDisplay.getLayoutRelativeURL();
+
+	if (layoutRelativeURL.includes('web')) {
+		return layoutRelativeURL.split('/').slice(0, 3).join('/');
+	}
+
+	return '';
+};
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.js:21]: Please, add a break line here.

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-undef */
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+const productID = fragmentElement
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.js:7]: Please, change the variable name to productId.
```suggestion
const productId = fragmentElement
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	.querySelector('.productID')
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.js:8]: You need to change this class name to product-id and if possible to become this selector more specific.


```suggestion
	.querySelector('.product-id')
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<div
+	class="d-none productID"
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/get-app-button/index.html:2]: Please, change the class name here.

```suggestion
	class="d-none product-id"
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/layout-page-templates/display-page-templates/app-detail/page-definition.json [hunk 5]
```diff
@@ -308,17 +308,23 @@
...
 											},
 											"fragmentConfig": {
+												"productId": {
+													"href": ""
+												}
 											},
 											"fragmentFields": [
 												{
-													"id": "element-html",
+													"id": "productID",
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/layout-page-templates/display-page-templates/app-detail/page-definition.json:321]: Please, change the id name.

```suggestion
													"id": "product-id",
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/layouts/07_get_app/page-definition.json [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+								"widthType": "Fixed"
+							}
+						},
+						"pageElements": [
+							{
+								"definition": {
+									"widgetInstance": {
+										"widgetConfig": {
+										},
+										"widgetName": "com_liferay_client_extension_web_internal_portlet_ClientExtensionEntryPortlet_20096_LXC_liferay_marketplace_custom_element_get_app"
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/layouts/07_get_app/page-definition.json:53]: There is a hard code value here in the widgetName's value. Please, use the replacement for that. Joana knows how to do that. Please, talk with her.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/CardButton/CardButton.tsx [hunk 7]
```diff
@@ -3,43 +3,49 @@
...
+        "card-button--disabled": disabled,
+        "card-button--selected": selected,
+      })}
+      onClick={onClick}
+    >
+      {icon ? (
+        icon
+      ) : (
+        <img alt="trial" className="card-button-icon" src={arrowLeft} />
+      )}
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/CardButton/CardButton.tsx:40]: Please, move this code to a function.

**Pull request closed without merging**